notes/dns-client-proxy.md

```diff
@@ -0,0 +1,535 @@
+
+The second option, called PROXY SCOPE Option, controls the IP address scope
+of the connection between the application's stub resolver and the proxy.
+
+Finally, the TRUST ANCHOR Option, provides the application with a DNSSEC
+trust anchor signed by IANA.
+
+
+# PROXY CONTROL OPTION
+
+~~~
+      +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+   0: |                          OPTION-CODE                          |
+      +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+   2: |                         OPTION-LENGTH                         |
+      +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+   4: | U |UA | A | P | D |DO |                     Z                 |
+      +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+   6: |A53|D53|AT |DT |AH2|DH2|AH3|DH3|AQ |DQ |         Z             |
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   8: |         Addr Type             |         Addr Length           |
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-|
+      ~                IPv4 or IPv6-address(es)                       ~
+      +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+      |  Domain Name Length           |                               |
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+                               |
+      ~                   Domain Name                                 ~
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+      |          SvcParams Length     |                               |
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+                               |
+      ~                 SvcParams                                     ~
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+      |     Interface Name Length     |                               |
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+                               |
+      ~                 Interface Name                                ~
+      +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+~~~
+
+where
+
+{newline="true"}
+OPTION-CODE
+
+: To be decided
+
+OPTION-LENGTH
+
+: Lenght of this option excluding the OPTION-CODE and OPTION-LENGTH fields
+
+U
+
+: force the use of unencrypted communication (Do53)
+
+UA
+
+: require unauthenticated encryption
+
+A
+
+: require authenticated encryption
+
+P
+
+: authenticate using a PKI certificate
+
+D
+
+: authenticate using DANE
+
+DO
+
+: disallow other transports (transports that are not explicitly listed)
+
+A53,AT,AH2,AH3,AQ
+
+: allow respectively Do53, DoT, DoH H2, DoH H3, DoQ
+
+D53,DT,DH2,DH3,DQ
+
+: disallow respectively Do53, DoT, DoH H2, DoH H3, DoQ
+
+Z
+
+: reserved, MUST be zero when sending, MUST be ignored when received
+
+Addr Type
+
+: Type of addresses, IPv4 or IPv6
+
+Addr Length
+
+: length of the addresses in octets. Must be a multiple of 4 for IPv4 and
+a multiple of 16 for IPv6
```

Comment:
Do we need to provision for scoped IPv6 addresses? I.e. like: fe80::1ff:fe23:4567:890a%eth2
I think we should somehow...

Reply:
We should make mention of this in the text, but my idea is that the interface field can be used to store for that

Reply:
I added text that for IPv6 link-local addresses the zone ID can be put in the interface field.